packages/backend/src/index.ts

```diff
@@ -69,6 +73,44 @@ export function initIocContainer(
     return new IORedis(config.redisUrl)
   })
 
+  /**
+   * Add services to the container.
+   */
+  container.singleton('userService', async (deps) => {
+    const logger = await deps.use('logger')
+    const knex = await deps.use('knex')
+    const accountService = await deps.use('accountService')
+    return await createUserService({
```

Comment:
a question about our use of dependency injection: Why do we construct the dependencies in a function here and instantiate the service with those passed in, rather than passing in the deps object and allowing the service to request all its dependencies from there? That's the pattern used in reduct, which is the dependency injection library we use in some other projects.

Reply:
The IoC container handles registration and resolution of service dependencies for us. By doing it this way we don't have to worry about the instantiation order of interdependent services.